lib/wick-get-iface-ip

```diff
@@ -20,36 +20,58 @@
 #
 # Returns 0 on success, non-zero on failure.
 wickGetIfaceIp() {
-    local ip list
+    local list target interface allInterfaces
 
-    if [[ "$2" == '*' ]]; then
-        list=()
+    wickGetArgument target 0 "$@"
+    wickGetArgument interface 1 "$@"
 
-        while read ip; do
-            ip=${ip#*inet addr:}
-            ip=${ip#*inet }
-            ip=${ip%% *}
-            list[${#list[@]}]=$ip
-        done < <(ifconfig -v 2>/dev/null | grep 'inet ' || echo "")
+    list=()
+    allInterfaces=false
 
+    if [[ "$interface" == "*" ]]; then
+        interface=""
+        allInterfaces=true
+    fi
+
+    if wickCommandExists ip; then
+        cmd=(ip addr show)
+
+        if [[ -n "$interface" ]]; then
+            cmd[${#cmd[@]}]="$interface"
+        fi
+
+        list=( $("${cmd[@]}" 2>/dev/null | grep -Po "inet\s+\K[\d.]+") )
+    elif wickCommandExists ifconfig; then
+        cmd=(ifconfig -v)
+
+        if [[ -n "$interface" ]]; then
+            cmd[${#cmd[@]}]="$interface"
+        fi
+
+        list=( $("${cmd[@]}" 2>/dev/null | grep -Po "inet\s+(addr:)?\K[\d.]+") )
+    else
+        wickError "Couldn't find a command to look up ip addresses"
+
+        return 2
+    fi
+
+    if [[ "$allInterfaces" == true ]]; then
         if [[ ${#list[@]} -eq 0 ]]; then
-            local "$1" && wickIndirectArray "$1"
+            local "$target" && wickIndirectArray "$target"
```

Comment:
This is a discussion point only.  It is inspired by the code review but would certainly not block this code from being accepted.

What do you think about this syntax?  I find it bulky.  I have a suggestion and I'm willing to implement it, but it certainly needs some thought and review by people (like you!).  Here's the suggestion:

* Remove the "if [[ ${#list[@]} -eq 0 ]]" block entirely.
* Change wickIndirectArray to contain that block.
* Add another parameter to wickIndirectArray so we pass the number of elements.

    local "$dest" && wickIndirectArray "$dest" ${#list[@]} "${list[@]}"

Perhaps we could do even more magic?  We could just pass the variable name and the scope of wickIndirectArray could, in theory, just grab from that variable.  We would need to remember to not use `$` in front.

    wickIndirectArray "$destinationVariable" list

Thoughts?  Alternatives?